### PR TITLE
Update gvisor.dev/gvisor to latest tagged release

### PR DESCRIPTION
**Pull request by cfergeau**
This will allow building with golang 1.20/1.21, but will break builds with
golang 1.19... If you need to build with 1.19, you will have to keep using
gvisor-tap-vsock 0.6.2

This fixes https://github.com/containers/gvisor-tap-vsock/issues/232

### CODE REVIEW

**Comment by cfergeau** [comment]
@AkihiroSuda this is https://github.com/containers/gvisor-tap-vsock/pull/219 rebased with an additional update on top of it

**Comment by lsm5** [comment]
We'll need to disable f37 packit builds for main branch as well. Could you please add this patch to update packit tasks?

```
$ cat 0001-update-packit-targets.patch 
From b951fd5b52a4623eb31f9fcb4cf702c57f1d26f2 Mon Sep 17 00:00:00 2001
From: Lokesh Mandvekar <lsm5@fedoraproject.org>
Date: Tue, 25 Jul 2023 15:32:56 -0400
Subject: [PATCH] update packit targets

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>
---
 .packit.yaml | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/.packit.yaml b/.packit.yaml
index 4322dda..8301f9e 100644
--- a/.packit.yaml
+++ b/.packit.yaml
@@ -18,9 +18,31 @@ jobs:
     enable_net: true
     srpm_build_deps:
       - make
+    targets:
+      - fedora-rawhide-aarch64
+      - fedora-rawhide-x86_64
+      - fedora-eln-aarch64
+      - fedora-eln-x86_64
+      - fedora-38-aarch64
+      - fedora-38-x86_64
+      - centos-stream-9-aarch64
+      - centos-stream-9-x86_64
+      - centos-stream-8-aarch64
+      - centos-stream-8-x86_64

   - <<: *copr
     # Run on commit to main branch
     trigger: commit
     branch: main
     project: podman-next
+    targets:
+      - fedora-rawhide-aarch64
+      - fedora-rawhide-x86_64
+      - fedora-eln-aarch64
+      - fedora-eln-x86_64
+      - fedora-38-aarch64
+      - fedora-38-x86_64
+      - centos-stream+epel-next-9-aarch64
+      - centos-stream+epel-next-9-ppc64le
+      - centos-stream+epel-next-9-s390x
+      - centos-stream+epel-next-9-x86_64
-- 
2.41.0

```

**Comment by cfergeau** [comment]
> We'll need to disable f37 packit builds for main branch as well. Could you please add this patch to update packit tasks?

The context for the diff you provided does not match what is in git main https://github.com/containers/gvisor-tap-vsock/blob/main/.packit.yaml
I've manually applied your changes and added them to this PR.

**Comment by lsm5** [comment]
@cfergeau @AkihiroSuda can we get this merged and cut a new release please?

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v3
       with:
-        go-version: 1.18
+        go-version: 1.20.x
```
Comment (praveenkumar) [inline .github/workflows/go.yml:20]: 👍🏼

**Review by praveenkumar — APPROVED** [review]

**Pull request merged**